openedx/core/djangoapps/content/block_structure/tasks.py

```diff
@@ -34,6 +34,7 @@ def update_course_in_cache(course_id):
         api.update_course_in_cache(course_key)
     except NO_RETRY_TASKS as exc:
         # Known unrecoverable errors
+        log.exception("update_course_in_cache encountered unrecoverable error in course {}".format(course_id))
```

Comment:
Recommend including the task id so we can correlate the message with the stack trace.  Add bind=True to the task decorator, give the function a `self` argument, and use `self.task_id`.